### ISSUE TITLE
Windows 11: NVDA renders a lot of spaces in the terminal with UIA enabled

### ISSUE THREAD

**Issue by brunoprietog** [nvaccess/nvda#13318]
### Steps to reproduce:

With UIA enabled, open a terminal in Windows 11, either WSL, CMD, Powershell or Windows Terminal.

Explore the output with the review cursor.

### Actual behavior:

On empty lines, if indentation is set to be announced, NVDA reports that there are 120 blanks.

On any other line, if you press the NVDA + end key, you should go to the last character on the line, but there are too many spaces. You have to go back with NVDA + left arrow many times to get to the last character.

If you use a plug-in like Virtual Review to make it more convenient to read the parts of the screen that are not visible, this problem is more annoying because of #13317.

### Expected behavior:

NVDA should not display this absurd amount of spaces. This problem does not occur on Windows 10 with UIA enabled.

#### NVDA installed/portable/running from source:

Installed

#### NVDA version:

2021.3.1

#### Windows version:

Windows 11 21H2 (x64) build 22000.469

#### Does the issue still occur after restarting your computer?

Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.

No

#### If NVDA add-ons are disabled, is your problem still occurring?

Yes

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?

Yes

**Comment by Brian1Gaff** [comment]
Windows versions are notorious for doing things differently.
 Even the command line works strangely sometimes.
 In Windows 7, if I run a batch file, half the time it never reads the press 
any key prompt if you have a pause in the file.
 Strange.
 Brian

***@***.***
Sent via blueyonder.
Please address personal E-mail to:-
***@***.***, putting 'Brian Gaff'
in the display name field.
Newsgroup monitored: alt.comp.blind-users
----- Original Message ----- 
From: "Bruno Prieto" ***@***.***>
To: "nvaccess/nvda" ***@***.***>
Cc: "Subscribed" ***@***.***>
Sent: Tuesday, February 08, 2022 9:37 AM
Subject: [nvaccess/nvda] Windows 11: NVDA renders a lot of spaces in the 
terminal with UIA enabled (Issue #13318)

**Comment by codeofdusk** [comment]
@brunoprietog Please press NVDA+f1 from a running conhost with UIA enabled. In the log viewer, what is the "API level" listed as?

**Comment by brunoprietog** [comment]
I get API level: 1 (IMPROVED)

**Comment by Neurrone** [comment]
I see this in Windows Terminal as well.

**Comment by codeofdusk** [comment]
This was changed in microsoft/terminal#4018 (i.e. when going from API level 0/`END_INCLUSIVE` to 1/`IMPROVED`). Before, terminal and conhost stripped *all* spaces from the end of lines, making Braille output incorrect among other things. Now, UIA console performs like legacy (i.e. shows all spaces present in the buffer). I recommend closing this.

CC @carlos-zamora.

**Comment by brunoprietog** [comment]
Why does removing the spaces cause problems for braille output? However, not removing the spaces also causes problems for the reasons I explained. It is very uncomfortable to press NVDA + End and not be able to get to the last character, as well as when copying the output or examining it with Virtual Review or the review cursor all those spaces appear. It was a great improvement that was with UIA in Windows 10.

**Comment by brunoprietog** [comment]
Consider #13317 and it's even worse.

**Comment by brunoprietog** [comment]
Oh, I'm sorry I don't have a braille line to test. I wonder if there is a way to fix both problems at the same time. So far, one problem is prioritized over the other, in my opinion, both are problems.

**Comment by codeofdusk** [comment]
Unfortunately, given the design of conhost, this is likely not possible. Showing all spaces is consistent with the behaviour of the legacy support (UIA off) since its beginning.

**Comment by Neurrone** [comment]
Having all spaces wasn't the behaviour in the past, at least before using UIA. It has caused problems in the past for me when the end of each line has dozens of unwanted spaces at the end.

Perhaps stripping spaces at the end of the line could be a setting that can be toggled for non braille users?

**Comment by brunoprietog** [comment]
I agree, this setting would be great, and that solves the problem for everyone.

**Comment by TheQuinbox** [comment]
With UIA enabled for Windows Console, I don't have this issue. Only when disabling it. Windows 11 insiders, build 22621

**Comment by Danstiv** [comment]
Any news?
